CMakeLists.txt

```diff
@@ -11,21 +19,20 @@ option(KokkosFFT_INTERNAL_Kokkos "Build internal Kokkos instead of relying on ex
 option(KokkosFFT_ENABLE_BENCHMARK "Build benchmarks for KokkosFFT" OFF)
 
 # Version information
-set(KokkosFFT_VERSION_MAJOR 0)
-set(KokkosFFT_VERSION_MINOR 0)
-set(KokkosFFT_VERSION_PATCH 00)
-set(KokkosFFT_VERSION "${KokkosFFT_VERSION_MAJOR}.${KokkosFFT_VERSION_MINOR}.${KokkosFFT_VERSION_PATCH}")
-
-#Set variables for config file
-math(EXPR KOKKOSFFT_VERSION "${KokkosFFT_VERSION_MAJOR} * 10000 + ${KokkosFFT_VERSION_MINOR} * 100 + ${KokkosFFT_VERSION_PATCH}")
+math(EXPR KOKKOSFFT_VERSION "${PROJECT_VERSION_MAJOR} * 10000 + ${PROJECT_VERSION_MINOR} * 100 + ${PROJECT_VERSION_PATCH} * 10 + ${PROJECT_VERSION_TWEAK}")
 math(EXPR KOKKOSFFT_VERSION_MAJOR "${KOKKOSFFT_VERSION} / 10000")
 math(EXPR KOKKOSFFT_VERSION_MINOR "${KOKKOSFFT_VERSION} / 100 % 100")
 math(EXPR KOKKOSFFT_VERSION_PATCH "${KOKKOSFFT_VERSION} % 100")
 
+set(KOKKOS_REQUIRED_VERSION "4.2.00")
+
 if (NOT KokkosFFT_INTERNAL_Kokkos)
     # First check, Kokkos is added as subdirectory or not
     if(NOT TARGET Kokkos::kokkos)
         find_package(Kokkos REQUIRED)
+        # Kokkos version check
```

Comment:
Maybe the Kokkos version check could be done unconditionally. I know it's unlikely that the embedded version comes outdated, but I think it doesn't hurt to have a straightforward approach.

Reply:
Is there a way to access internal CMake variables in Kokkos CMake?

Reply:
`find_package` can check for version directly no ?

Reply:
If we rely on a pre-installed Kokkos yes. I am not sure how can I retrieve information from Kokkos added as a subdirectory

Reply:
I tested it with the sub-directory way and could access `Kokkos_VERSION`.

Reply:
So, it would work except for using Kokkos as subdirectory of this `KokkosFFT`?